packages/dropdowns/src/styled/field/StyledSelect.ts

```diff
@@ -17,72 +17,83 @@ const isInvalid = (validation?: VALIDATION) => {
   return validation === 'warning' || validation === 'error';
 };
 
-interface IStyledSelectIconProps {
-  isCompact?: boolean;
-  isBare?: boolean;
-}
+const iconStyles = (props: IStyledSelectIconProps & ThemeProps<DefaultTheme>, isStart: boolean) => {
+  const maxHeight = `${props.theme.space.base * (props.isCompact ? 8 : 10)}px`;
+  const padding = `${props.theme.space.base * 3}px`;
+  const size = props.theme.iconSizes.md;
+  let position;
+  let justifyContent;
+  let shade;
+
```

Comment:
This is crafty. It took me like 20 minutes to understand why this is working but I think I finally got it. I kept thinking it was redundant. The RTL logic makes my brain hurt.